src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -339,12 +341,21 @@ def _search_in_extension_commands(self, command_str):
 
     def _get_extension_use_dynamic_install_config(self):
         cli_ctx = self.cli_ctx or (self.cli_help.cli_ctx if self.cli_help else None)
+        default_value = 'yes_prompt'
         use_dynamic_install = cli_ctx.config.get(
-            'extension', 'use_dynamic_install', 'no').lower() if cli_ctx else 'no'
+            'extension', 'use_dynamic_install', default_value).lower() if cli_ctx else default_value
         if use_dynamic_install not in ['no', 'yes_prompt', 'yes_without_prompt']:
```

Comment:
`yes_prompt` seems redundant to me. `prompt` should be enough.

Reply:
`yes_prompt` and `yes_without_prompt` originate from the `yes` option, let's keep it there to be explicit and backward compatible.